src/test/java/picard/sam/markduplicates/MarkDuplicatesTagRepresentativeReadIndexTester.java

```
@@ -0,0 +1,126 @@
+import org.testng.Assert;
+import picard.PicardException;
+import picard.cmdline.CommandLineProgram;
+import picard.sam.DuplicationMetrics;
+
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class is an extension of AbstractMarkDuplicatesCommandLineProgramTester used to test MarkDuplicatesWithMateCigar with SAM files generated on the fly.
+ * This performs the underlying tests defined by classes such as see AbstractMarkDuplicatesCommandLineProgramTest and MarkDuplicatesWithMateCigarTest.
+ * @author hogstrom@broadinstitute.org
+ */
+public class MarkDuplicatesTagRepresentativeReadIndexTester extends AbstractMarkDuplicatesCommandLineProgramTester {
+
+    //final public Map<String, String> readToRepReadMap = new HashMap<>(); // map to contain representative read names for each record
+    final public Map<Integer, Integer> expectedRepresentativeIndexMap = new HashMap<>();
+    final public Map<String, Integer> expectedSetSizeMap = new HashMap<>();
+    public boolean testRepresentativeReads = false;
+
+    public MarkDuplicatesTagRepresentativeReadIndexTester() {
+
+        addArg("TAGGING_POLICY=All");
+        addArg("TAG_DUPLICATE_SET_MEMBERS=true");
+    }
+
+    @Override
+    protected CommandLineProgram getProgram() { return new MarkDuplicates(); }
+
+    @Override
+    public void test() {
+        try {
+            updateExpectedDuplicationMetrics();
+            // Read the output and check the duplicate flag
+            int outputRecords = 0;
+            int indexInFile = 0;
+            final SamReader reader = SamReaderFactory.makeDefault().open(getOutput());
+            for (final SAMRecord record : reader) {
+                outputRecords++;
+
+                final String key = samRecordToDuplicatesFlagsKey(record);
+                if (!this.duplicateFlags.containsKey(key)) {
+                    System.err.println("DOES NOT CONTAIN KEY: " + key);
```

Instead of printing to stderr, would it be better to throw an exception?